app-shopping/src/main/java/com/shopping/service/PhoneServiceImpl.java

```diff
@@ -47,4 +49,23 @@ private PhoneModel updateColor(PhoneModel modle) {
         return modle;
     }
 
+    public Long addPhone(PhoneModel pm)
+    {
+        return modelRepository.save(pm).getId();
+    }
+
+    public void deletePhone(String modelPrice)
+    {
+        modelRepository.deleteBymodelPrice(modelPrice);
+    }
+
+    @Override
+    public List<PhoneModel> findByModelColour(String modelColour) {
+        List<PhoneModel> pm = modelRepository.findByModelColour(modelColour);
+        if (!pm.isEmpty()) {
```

Comment:
use pm.isPresent()

Reply:
If i use Optional<List<PhoneModel>> its returning the **Optional[[]]**  value and showing some value is present and going to the if block even if the modelColour is not present.